### ISSUE TITLE
[CI] Pin edge build docker containers

### ISSUE THREAD

**Issue by leezu** [apache/incubator-mxnet#17151]
## Description
Currently the pinned containers used in the edge build are broken and things only work due to caching effects https://github.com/apache/incubator-mxnet/issues/16753
However, caching breaks in case of any changes to the edge build.
Thus https://github.com/apache/incubator-mxnet/issues/16753, which requires a change (cmake update) disabled pinning and switched to the upstream containers.

However, based on https://github.com/apache/incubator-mxnet/pull/17063#issuecomment-566808371 it's good to reintroduce pinning.

**Comment by samskalicky** [comment]
@apeforest assign [@larroy ]

**Comment by apeforest** [comment]
@larroy please assign yourself.

**Comment by larroy** [comment]
@apeforest how should I got about doing that?

**Comment by szha** [comment]
@apeforest non-committers need to comment on an issue before they can be assigned.
